src/sentry/models/avatar.py

```diff
@@ -33,9 +33,25 @@ def save(self, *args, **kwargs):
             self.ident = uuid4().hex
         return super().save(*args, **kwargs)
 
+    def get_file(self):
+        from sentry.models import File
+
+        if self.file_id is None:
+            return None
+
+        try:
+            return File.objects.get(pk=self.file_id)
+        except File.DoesNotExist:
+            # Best effort replication of previous behaviour with foreign key
+            # which was set with on_delete=models.SET_NULL
```

Comment:
We should probably make a query and do this actively when we delete a file as well. We could have a `post_delete` roughly like

```
def post_delete(...):
    models = (UserAvatar, OrganizationAvatar, TeamAvatar, ProjectAvatar, ...)
    for model in models:
        model.objects.filter(file_id=file_id).update(file_id=None)
```

Reply:
I don't think it's necessary, looking at the avatar api/model code, there _should_ be no way to remove the avatar associated file without updating the avatar object as well. If I read it correctly atm it's not even possible to have avatar without a file (`sentry_useravatar` table says something else).

But mostly I don't think it's a good idea. This would multiply the queries we do for file deletion 5 times and we do a lot of file deletion, especially as part of the cleanup job.

If you still think that setting `file_id` to `None` on retrieval is not enough and we should do it proactively, I'd go either with a scheduled task or at least try to batch the update queries somehow.

Reply:
> But mostly I don't think it's a good idea. This would multiply the queries we do for file deletion 5 times and we do a lot of file deletion, especially as part of the cleanup job.

It's how things are handled now - the Django ORM will do these queries already as part of deleting the model, we'd just be manually implementing the same process since Django no longer knows about the relationships. Django doesn't actually use database level cascades at all, it implements these things at the application level.

https://github.com/django/django/blob/f1e3017aeaeddc590dcf2cf88511f3a726da73ca/django/db/models/deletion.py#L219
https://github.com/django/django/blob/f1e3017aeaeddc590dcf2cf88511f3a726da73ca/django/db/models/deletion.py#L230

One thing that would be worse is if we're bulk deleting files in one batch. If that's the case, then it'll be a lot worse. I don't think this matters a lot, just seems better to handle things proactively if we can.

Reply:
Discussed this on slack, we'll move to not updating the related models since it's not too important, and also saves us some extra queries